tests/integration/test_task.py

```diff
@@ -50,7 +31,29 @@ def _run(self, input_files):
 
         return [output_file]
 
-    monkeypatch.setattr(BaseTask, '_run', _run)
+
+@pytest.fixture
+def example_tasks(tmp_path):
+    """Example tasks for testing the task runners."""
+    tasks = TaskSet()
+    for i in range(3):
+        task = MockBaseTask(
+            name=f'task{i}',
+            ancestors=[
+                MockBaseTask(name=f'task{i}-ancestor{j}') for j in range(3)
+            ],
+        )
+        for task0 in task.flatten():
+            task0.priority = i
+            task0._tmp_path = tmp_path
+        tasks.add(task)
+
+    return tasks
+
+
+@pytest.mark.parametrize('max_parallel_tasks', [1, 2, 3, 4, 16, None])
+def test_run_tasks(monkeypatch, tmp_path, max_parallel_tasks, example_tasks):
+    """Check that tasks are run correctly."""
 
     example_tasks.run(max_parallel_tasks=max_parallel_tasks)
```

Comment:
Maybe add a parameterizer with 'fork' / 'spawn' that does something like this:

```suggestion
    monkeypatch.setattr(esmvalcore._task, 'Pool', mp.get_context('spawn').Pool)
    example_tasks.run(max_parallel_tasks=max_parallel_tasks)
```